packages/gateway/src/features/user/type-resolvers.ts

```diff
@@ -207,6 +211,10 @@ export const userTypeResolvers: GQLResolver = {
 
       const signatureExtension = getSignatureExtension(practitioner.extension)
 
+      if (userModel.role === 'FIELD_AGENT') {
+        return null
+      }
+
```

Comment:
Wouldn't it be better to check the role before the fetching the practitioner? That way for FieldAgents we will have one less call to FHIR.